src/app/services/stomp/stomp.service.ts

```diff
@@ -23,53 +24,55 @@ export enum STOMPState {
  * message queue using the stomp.js library, and returns
  * values via the ES6 Observable specification for
  * asynchronous value streaming by wiring the STOMP
- * messages into a Subject observable.
+ * messages into a observable.
  */
 @Injectable()
 export class STOMPService {
 
-  /* Service parameters */
-
   // State of the STOMPService
   public state: BehaviorSubject<STOMPState>;
 
-  // Publishes new messages to Observers
-  public messages: Subject<Stomp.Message>;
+  // Will trigger when connection is established, will trigger immediately once if already connected
+  public connectObservable: Observable<number>;
 
   // Configuration structure with MQ creds
   private config: StompConfig;
 
   // STOMP Client from stomp.js
   private client: Stomp.Client;
 
-  // Resolve Promise made to calling class, when connected
-  private resolvePromise: (...args: any[]) => void;
-
-  // Timer
-  private timer: NodeJS.Timer;
-
   /** Constructor */
-  public constructor() {
-    this.messages = new Subject<Stomp.Message>();
+  public constructor(private _configService: ConfigService) {
     this.state = new BehaviorSubject<STOMPState>(STOMPState.CLOSED);
-  }
 
+    this.connectObservable = this.state
+      .filter((currentState: number) => {
+        return currentState === STOMPState.CONNECTED
+      });
+
+    // Setup sending queuedMessages
+    this.connectObservable.subscribe(() => {
+      // The delay is just for testing during development, as I am publishing and subscribing same Stomp topic. the delaye
```

Comment:
Delay for testing during development: was it your intention to include this as part of the PR? 

It sounds like perhaps this should be a permanent feature?